src/server/database/Database/PreparedStatement.cpp

```diff
@@ -82,112 +82,169 @@ void PreparedStatement::BindParameters()
 }
 
 //- Bind to buffer
-void PreparedStatement::setBool(const uint8 index, const bool value)
+void PreparedStatement::setBool(const uint8 index, const bool value, bool setNullIfZero)
 {
-    if (index >= statement_data.size())
-        statement_data.resize(index+1);
-
-    statement_data[index].data.boolean = value;
-    statement_data[index].type = TYPE_BOOL;
+    if (setNullIfZero && !value)
+        setNull(index);
+    else
+    {
+        if (index >= statement_data.size())
+            statement_data.resize(index+1);
```

Comment:
missing linebreak after `statement_data.resize(index+1);` and missing whitespace between `index+1`
-> also for all following and previous changes in this PR.